### PR TITLE
ytplayer.config fallback

### PR DESCRIPTION
**Pull request by mchangrh**
taken from #1159

### CODE REVIEW

File: src/content.ts [hunk 1]
```diff
@@ -999,6 +999,16 @@ function getYouTubeVideoIDFromURL(url: string): string | boolean {
             return false;
         }
     }
+
+    interface ytPlayerWindow extends Window {
```
Comment (ajayyy) [inline src/content.ts:1003]: It won't get here since searchParams has v, but id.length is not 11

Comment (mchangrh) [reply src/content.ts]: moved to top, since it should be the most reliable if it does exist

**Comment by ajayyy** [comment]
Todo: test how fast this changes. Does url change first?

**Comment by mchangrh** [comment]
put it in URL search since it's only applicable on the youtube site, there's also ytInitialData which might be another fallback

**Comment by mchangrh** [comment]
also I think window.ytplayer might be a/b tested since i'm not seeing it in private window, that or it's only on specific pages

**Comment by ajayyy** [comment]
ytInitialData only is valid for the first video, but stays in the page, so I don't think it's a good idea to rely on

**Comment by mchangrh** [comment]
I misunderstood

**Comment by mchangrh** [comment]
seems like ytcfg might also be a reliable source of videoID

**Comment by mchangrh** [comment]
nope, window.ytplayer follows ytinitialdata in playlists

**Pull request closed without merging**